docs/asciidoc/tcp-client.adoc

```diff
@@ -342,3 +343,70 @@ include::{examplesdir}/uds/Application.java[lines=18..33]
 ----
 <1> Specifies `DomainSocketAddress` that will be used
 ====
+
+== Host Name Resolution
+By default, the `TCP` client uses Netty's domain name lookup mechanism that resolves a domain name asynchronously.
```

Comment:
same as for the http client, use `TcpClient` class name instead of `TCP client`?